### ISSUE TITLE
Issues with failure_policy_move_dest_dir (when failure_policy is 'move')

### ISSUE THREAD

**Issue by dearith** [metwork-framework/acquisition#12]
Hi,

I'm trying to use `failure_policy_move_dest_dir`  with `failure_policy ` (`move`).

If think there is an issue in step.py (line 65 : mkdir_p_or_die(fpmdd)):

```
        if self.failure_policy == "move":
            fpmdd = self.args.failure_policy_move_dest_dir
            if fpmdd is None or fpmdd == "" or fpmdd == "FIXME":
                self.error_and_die(
                    "you have to set a "
                    "failure-policy-move-dest-dir"
                    " in case of move failure policy"
                )
            if "/" not in fpmdd or fpmdd.startswith('/'):
                self.error_and_die(
                    "failure-policy-move-dest-dir must be something like "
                    "plugin_name/step_name")
            mkdir_p_or_die(fpmdd)
            plugin_name = fpmdd.split('/')[0]
            step_name = fpmdd.split('/')[1]
            self.add_virtual_trace(plugin_name, step_name)
            self.failure_policy_move_dest_dir = \
                get_plugin_step_directory_path(plugin_name, step_name)

```

The final failure_policy_move_dest_dir is step.xxx.yyy and in line 65, we are trying to create the directory xxx/yyy.

So If I set failure_policy_move_dest=xxx/yyy. The directory step.xxx.yy doesn't exist and the file won't be copied.

There is something I don't understand :why the failure_policy_move_dest_dir must be something like xxx/yyy ?

In Metwork 0.9, we are free to set the directory we want.

So, I no longer understand the use case of the move failure policy.  

My use case may be something like : if the file can't be processed, I want to copy it in a specific directory in order to debug and/or inspect what's wrong and potentially process again the file.

**Comment by thefab** [comment]
`xxx/yyy` means `{plugin_name}/{step_name}`

I have to check this but if you want to use another directory not in this scheme, you can use an absolute value (starting with /)

**Comment by dearith** [comment]
I can't use an absolute value (starting with /) : because of the check :

```
 if "/" not in fpmdd or fpmdd.startswith('/'):
                self.error_and_die(
                    "failure-policy-move-dest-dir must be something like "
                    "plugin_name/step_name")
```
So, if it starts with '/', the error is raised : "failure-policy-move-dest-dir must be something like plugin_name/step_name"

**Comment by dearith** [comment]
Notice I think there is a second issue :  `mkdir_p_or_die(fpmdd)` at line 65 creates the directories `xxx/yyy` under the plugin directory, I don't think this directory is used. When `failure_policy_move_dest=xxx/yyy`, the directory used is `step.xxx.yyy` under the `<MFMODULE_RUNTIME_HOME>/var/in` directory.

**Comment by thefab** [comment]
ok there is at least one bug

before the fix, you can install  and configure "move" plugin (and configure failure policy to send to this plugin)

**Comment by dearith** [comment]
Thanks, that's what I did, we are using our own plugin "move" based on the "move" plugin. 
The advantage of the "move" plugin is that we cant save (move) the input file with its original name (dest_basename + timestamp).

We have cloned the "move" plugin, because we have more than one step in our move plugin (we want distinct directories for each type of data we process in mfdata).

That works fine without any additional code, thanks Metwork 👍
